src/lib/core/CHIPError.h

```diff
@@ -392,6 +424,16 @@ class [[nodiscard]] ChipError
     {
         return MakeInteger(Range::kSDK, MakeField(kSdkPartStart, to_underlying(part)) | MakeField(kSdkCodeStart, code));
     }
+
+    static constexpr StorageType MaskValue(Range range, ValueType value)
+    {
+        // Mask value for every range except kPlatform range. The kPlatform range
+        // is special, because we are using only the highest bit to determine whether
+        // the range is kPlatform and the rest is used to store the value itself.
```

Comment:
```suggestion
        // the range is kPlatform and the rest is used to store the value itself.
        // No masking is needed for kPlatform because MakeInteger will set the MSB.
```